packages/quantification/src/net-quantification.ts

```diff
@@ -0,0 +1,185 @@
+    UnadjustedQuantificationSummary,
+    'somscAnnualDifferencesBetweenFutureAndBaselineScenarios'
+  >[],
+  logger?: Pick<Console, 'debug' | 'table'>
+): AnnualTotals[] => {
+  const netQuantifications: AnnualTotals[] = quantifications.map(
+    (quantification) => ({
+      ...quantification.somscAnnualDifferencesBetweenFutureAndBaselineScenarios,
+    })
+  );
+
+  // Get the set of years in all quantifications
+  const years = new Set<string>();
+  for (const quantification of netQuantifications) {
+    for (const year of Object.keys(quantification)) {
+      years.add(year);
+    }
+  }
+
+  /**
+   * CAUTION -- years are represented as strings and sorted alphabetically. If for some reason, we
+   * begin supporting years < 1000 or > 9999, we'll need to rework this sort
+   */
+  const yearsOrderedAsc = Array.from(years.values()).sort();
+  logger?.debug('All years:', yearsOrderedAsc);
+
+  // No guarantee that all fields have data for each year, so we need to add zeroes for the missing years
+  for (const quantification of netQuantifications) {
+    for (const year of years) {
+      if (quantification[year] === undefined) {
+        quantification[year] = 0;
+      }
+    }
+  }
+
+  let rowIndex = 0;
+  let colIndex = 0;
+  let debt = 0;
+  while (rowIndex < netQuantifications.length) {
+    while (colIndex < yearsOrderedAsc.length) {
+      const yearIndex = yearsOrderedAsc[colIndex];
+      const value = netQuantifications[rowIndex][yearIndex];
+      logger?.debug(`Visiting [${rowIndex}, ${yearIndex}] = ${value}`);
+
+      if (value < 0) {
+        logger?.debug(`Found negative number (${value}), adding to debt`);
+        debt += value;
+        netQuantifications[rowIndex][yearIndex] = 0;
+      }
+
+      if (debt < 0) {
+        logger?.debug(`Accounting for debt of ${debt}`);
+        // Edge case: if there's only one year, start on the next field, otherwise start on the current
+        let accountingRowIndex =
+          yearsOrderedAsc.length === 1
+            ? rowIndex + (1 % netQuantifications.length)
+            : rowIndex;
+        let accountingColIndex = (colIndex + 1) % years.size;
+        let accountingYearIndex = yearsOrderedAsc[accountingColIndex];
+        while (debt < 0) {
+          const accountingCellValue =
+            netQuantifications[accountingRowIndex][accountingYearIndex];
+
+          logger?.debug(
+            `Accounting visiting [${accountingRowIndex}, ${accountingYearIndex}] = ${accountingCellValue}`
+          );
+
+          // Core logic: the amount to take is the lesser of the absolute value of the debt or the
+          // the amount available
+          const amountAvailable = Math.max(accountingCellValue, 0);
+          const amountToTake = Math.min(amountAvailable, Math.abs(debt));
+          const cellRemainder = accountingCellValue - amountToTake;
+          netQuantifications[accountingRowIndex][accountingYearIndex] =
+            cellRemainder;
+          debt += amountToTake;
```

Comment:
can you make sure to use our math library for all math operations? Otherwise we risk relying on javascripts [floating point arithmetic problems
](https://mathjs.org/docs/datatypes/bignumbers.html#roundoff-errors)
e.g.

```js 
1.1 + 1.3; // 2.4000000000000004
```

Reply:
Yeah, will change.